src/test/java/com/github/cameltooling/lsp/internal/documentsymbol/DocumentSymbolProcessorTest.java

```diff
@@ -59,12 +64,41 @@ void testRoutesProvidedAsDocumentSymbol() throws Exception {
 				"      </recipientList>\r\n" + 
 				"    </route>\n"
 				+ "</camelContext>\n";
-		List<Either<SymbolInformation, DocumentSymbol>> documentSymbols = testRetrieveDocumentSymbol(textTotest, 3);
-		SymbolInformation firstRoute = documentSymbols.get(0).getLeft();
-		assertThat(firstRoute.getName()).isEqualTo("a route");
+		List<Either<SymbolInformation, DocumentSymbol>> documentSymbols = testRetrieveDocumentSymbol(textTotest, 8);
+		SymbolInformation firstRoute = documentSymbols.stream()
+			.filter(item -> "a route".equals(item.getLeft().getName()))
+			.map(item -> item.getLeft())
+			.findAny()
+			.get();
 		Position expectedStart = new Position(2, 24/* expecting 4 but seems a bug in Camel*/);
 		Position expectedEnd = new Position(8, 12);
-		assertThat(firstRoute.getLocation()).usingRecursiveComparison().isEqualTo(new Location(DUMMY_URI+".xml", new Range(expectedStart, expectedEnd)));
+		assertThat(firstRoute.getLocation().getRange()).usingRecursiveComparison().isEqualTo(new Range(expectedStart, expectedEnd));
+	}
+	
+	@Test
+	void testEndpointsInXml() throws Exception {
+		String textTotest =
+				"<camelContext id=\"camel\" xmlns=\"http://camel.apache.org/schema/spring\">\r\n" +
+				"    <route id=\"a route\">\r\n" +
+				"      <from uri=\"direct:cafe\"/>\r\n" +
+				"      <to uri=\"direct:drink\"/>\r\n" +
+				"    </route>\r\n" +
+				"</camelContext>\n";
+		List<Either<SymbolInformation, DocumentSymbol>> documentSymbols = testRetrieveDocumentSymbol(textTotest, 4);
+		checkSymbolInformation(documentSymbols, new Position(2, 0), new Position(3, 30), "from direct:cafe");
```

Comment:
this test is where we can see that the "from" incorporates all CamelNodes. This is what allows to have the hierarchy in outline, similar to what is done in Java. The drawback is when using the "search document symbol"

end position to be replaced with (2,32) if we want to go the other way.